AzureRM plugin crashes when trying to update `azure_rm_publicip` in-place

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

v1.9.3

### AzureRM Provider Version

v3.113.0

### Affected Resource(s)/Data Source(s)

azurerm_public_ip

### Terraform Configuration Files

```hcl
resource "azurerm_public_ip" "location_frontend_public_ip_prx01" {
  name                = "prx01-public-ip"
  location            = azurerm_resource_group.rg_location.location
  resource_group_name = azurerm_resource_group.rg_location.name
  allocation_method   = "Static"
  sku                 = "Standard"
  ip_version          = "IPv4"
  idle_timeout_in_minutes = "30"
  tags = {
    role = "network"
    environment = "production"
  }
}
        
resource "azurerm_public_ip" "location_frontend_public_ip_sql02" {
  name                = "sql02-public-ip"
  location            = azurerm_resource_group.rg_location.location
  resource_group_name = azurerm_resource_group.rg_location.name
  allocation_method   = "Static"
  sku                 = "Standard"
  ip_version          = "IPv4"
  idle_timeout_in_minutes = "30"
  tags = {
    role = "network"
    environment = "production"
  }
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.113.0_x5 plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 100 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourcePublicIpUpdate(0xc001731980, {0x7245380?, 0xc001478900})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/public_ip_resource.go:347 +0xf58
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x8a558c0?, {0x8a558c0?, 0xc00248f8c0?}, 0xd?, {0x7245380?, 0xc001478900?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0018d88c0, {0x8a558c0, 0xc00248f8c0}, 0xc0014fcf70, 0xc001731680, {0x7245380, 0xc001478900})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000a72f78, {0x8a558c0?, 0xc00248f7d0?}, 0xc0011d2280)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001ddaaa0, {0x8a558c0?, 0xc00248efc0?}, 0xc0017cd340)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7ed90e0?, 0xc001ddaaa0}, {0x8a558c0, 0xc00248efc0}, 0xc0017cd2d0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000038000, {0x8a805c0, 0xc001f2e9c0}, 0xc0024907e0, 0xc001ef7ce0, 0xe081498, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc000038000, {0x8a805c0, 0xc001f2e9c0}, 0xc0024907e0, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 16
        google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-azurerm_v3.113.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

Terraform should have been able to update my Public IP objects in place with my new configuration. I changed `idle_timeout_in_minutes` from the default of 4 to 30 minutes.

These were the staged changes:

```
Terraform will perform the following actions:

  # azurerm_public_ip.location_frontend_public_ip_prx01 will be updated in-place
  ~ resource "azurerm_public_ip" "location_frontend_public_ip_prx01" {
        id                      = "/subscriptions/azure-subscription-id/resourceGroups/location-rg/providers/Microsoft.Network/publicIPAddresses/prx01-public-ip"
      ~ idle_timeout_in_minutes = 4 -> 30
        name                    = "prx01-public-ip"
        tags                    = {
            "environment" = "production"
            "role"        = "network"
        }
        # (11 unchanged attributes hidden)
    }

  # azurerm_public_ip.location_frontend_public_ip_sql02 will be updated in-place
  ~ resource "azurerm_public_ip" "location_frontend_public_ip_sql02" {
        id                      = "/subscriptions/azure-subscription-id/resourceGroups/location-rg/providers/Microsoft.Network/publicIPAddresses/sql02-public-ip"
      ~ idle_timeout_in_minutes = 4 -> 30
        name                    = "sql02-public-ip"
        tags                    = {
            "environment" = "production"
            "role"        = "network"
        }
        # (11 unchanged attributes hidden)
    }

Plan: 0 to add, 2 to change, 0 to destroy.
```

### Actual Behaviour

I received this error message upon utilizing `terraform apply` with the resulting `tfplan` that I generated via `terraform plan`:

```
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_public_ip.location_frontend_public_ip_prx01,
│   on frontend_virtual_machine_public_ips.tf line 2, in resource "azurerm_public_ip" "location_frontend_public_ip_prx01":
│    2: resource "azurerm_public_ip" "location_frontend_public_ip_prx01" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_public_ip.location_frontend_public_ip_sql02,
│   on frontend_virtual_machine_public_ips.tf line 21, in resource "azurerm_public_ip" "location_frontend_public_ip_sql02":
│   21: resource "azurerm_public_ip" "location_frontend_public_ip_sql02" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵
```

This was followed by the "Debug Output/Panic Output" I shared above.

### Steps to Reproduce

1. Create public IP and have it's `idle_timeout_in_minutes` as the default.
2. Change that field's value to 30 or something else (I have only reproduced with my current change)
3. Save your file, `terraform plan -out=tfplan` and then `terraform apply 'tfplan'`
4. Observe the issue?

### Important Factoids

I was able to reproduce in AzureRM provider version `3.106.1` as well. I upgraded to latest to try to see if it was still present, and it was. We use Azure pay as you go for our infrastructure.

### References

I cannot find anything related to my specific issue.

--- a/internal/services/network/public_ip_resource.go
+++ b/internal/services/network/public_ip_resource.go
@@ -344,7 +344,7 @@ func resourcePublicIpUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 	}
 
 	if d.HasChange("idle_timeout_in_minutes") {
-		payload.Properties.IdleTimeoutInMinutes = utils.Int64(d.Get("idle_timeout_in_minutes").(int64))
+		payload.Properties.IdleTimeoutInMinutes = utils.Int64(int64(d.Get("idle_timeout_in_minutes").(int)))
 	}
 
 	if d.HasChange("domain_name_label") {

--- a/internal/services/network/public_ip_resource_test.go
+++ b/internal/services/network/public_ip_resource_test.go
@@ -740,12 +740,13 @@ resource "azurerm_resource_group" "test" {
 }
 
 resource "azurerm_public_ip" "test" {
-  name                = "acctestpublicip-%d"
-  location            = azurerm_resource_group.test.location
-  resource_group_name = azurerm_resource_group.test.name
-  allocation_method   = "Static"
-  sku                 = "Basic"
-  domain_name_label   = "acctest-%d"
+  name                    = "acctestpublicip-%d"
+  location                = azurerm_resource_group.test.location
+  resource_group_name     = azurerm_resource_group.test.name
+  allocation_method       = "Static"
+  sku                     = "Basic"
+  domain_name_label       = "acctest-%d"
+  idle_timeout_in_minutes = 30
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
 }

--- a/internal/services/network/public_ip_resource_test.go
+++ b/internal/services/network/public_ip_resource_test.go
@@ -238,10 +238,16 @@ func TestAccPublicIpStatic_idleTimeout(t *testing.T) {
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.idleTimeout(data),
+			Config: r.idleTimeout(data, 30),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
-				check.That(data.ResourceName).Key("idle_timeout_in_minutes").HasValue("30"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.idleTimeout(data, 15),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
 		data.ImportStep(),
@@ -751,7 +757,7 @@ resource "azurerm_public_ip" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
 }
 
-func (PublicIPResource) idleTimeout(data acceptance.TestData) string {
+func (PublicIPResource) idleTimeout(data acceptance.TestData, idleTimeout int) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
   features {}
@@ -768,9 +774,9 @@ resource "azurerm_public_ip" "test" {
   resource_group_name     = azurerm_resource_group.test.name
   allocation_method       = "Static"
   sku                     = "Basic"
-  idle_timeout_in_minutes = 30
+  idle_timeout_in_minutes = %d
 }
-`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, idleTimeout)
 }
 
 func (PublicIPResource) dynamic_basic(data acceptance.TestData) string {

